test/functional/api/v2/compute_resources_controller_test.rb

```diff
@@ -148,6 +150,43 @@ def teardown
     assert !available_storage_domains.empty?
   end
 
+  test "should get available availability flavors" do
+    flavor = Object.new
+    flavor.stubs(:name).returns('test_flavor')
+    flavor.stubs(:id).returns('123')
+
+    Foreman::Model::EC2.any_instance.stubs(:available_flavors).returns([flavor])
+
+    get :available_flavors, { :id => compute_resources(:ec2).to_param }
+    assert_response :success
+    available_flavors = ActiveSupport::JSON.decode(@response.body)
+    assert_not_empty available_flavors
+  end
+
+  test "should get available security groups" do
```

Comment:
These tests look somewhat repetitive. It seems that the only thing that changes is the name of the method, `flavors`, `security_groups`, `availability_zones`. 

Would you mind to [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself) this up a bit? Making some method that runs these tests with a couple of arguments `compute_resource, fog_method` or something similar should be fine. 

The tests in the lines above these new ones would also fit in a helper method like that, but I can't ask you to fix others' code, feel free to make them use the refactor though :smile: